pkg/validation/config.go

```diff
@@ -24,13 +24,19 @@ type Validator struct {
 
 // NewValidator creates an object that optimizes bulk validations.
 func NewValidator() Validator {
-	profiles := api.ClusterProfiles()
+	return Validator{
+		hasTrapCache: make(map[string]bool),
+	}
+}
+
+// NewCPValidator creates an object that  and optimizes bulk validations.
+func NewCPValidator(profiles api.ClusterProfilesList) Validator {
```

Comment:
I basically split the function into 2, because the `validClusterProfiles` field of `Validator` is not filled anywhere besides when it is used for Cluster Profile checking.

Reply:
Are there any cases where we'd want to create a `Validator` that does not check for profile owners?

If not, we should have only one `New` function and modify all callers.

Reply:
@psalajova I am also interested to know if there are other valid usages without validating cluster profiles